vendor/bat-native-ads/src/bat/ads/internal/processors/contextual/text_embedding/text_embedding_processor.cc

```diff
@@ -0,0 +1,126 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_processor.h"
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_html_events.h"
+
+#include <algorithm>
+#include <vector>
+
+#include "base/check.h"
+#include "bat/ads/internal/base/logging_util.h"
+#include "bat/ads/internal/base/search_engine/search_engine_results_page_util.h"
+#include "bat/ads/internal/base/search_engine/search_engine_util.h"
+#include "bat/ads/internal/features/text_embedding_features.h"
+#include "bat/ads/internal/locale/locale_manager.h"
+#include "bat/ads/internal/ml/pipeline/text_processing/embedding_data.h"
+#include "bat/ads/internal/ml/pipeline/text_processing/embedding_processing.h"
+#include "bat/ads/internal/resources/contextual/text_embedding/text_embedding_resource.h"
+#include "bat/ads/internal/resources/language_components.h"
+#include "bat/ads/internal/resources/resource_manager.h"
+#include "bat/ads/internal/tabs/tab_manager.h"
+#include "url/gurl.h"
+
+namespace ads {
+namespace processor {
+
+TextEmbedding::TextEmbedding(resource::TextEmbedding* resource)
+    : resource_(resource) {
+  DCHECK(resource_);
+
+  LocaleManager::GetInstance()->AddObserver(this);
+  ResourceManager::GetInstance()->AddObserver(this);
+  TabManager::GetInstance()->AddObserver(this);
+}
+
+TextEmbedding::~TextEmbedding() {
+  LocaleManager::GetInstance()->RemoveObserver(this);
+  ResourceManager::GetInstance()->RemoveObserver(this);
+  TabManager::GetInstance()->RemoveObserver(this);
+}
+
+bool TextEmbedding::IsEmbeddingEnabled() {
+  return targeting::features::IsTextEmbeddingEnabled();
+}
+
+void TextEmbedding::Process(const std::string& text) {
+  if (!resource_->IsInitialized()) {
+    BLOG(1, "Failed to process token embeddings as resource not initialized");
+    return;
+  }
+
+  ml::pipeline::EmbeddingProcessing* embedding_proc_pipeline = resource_->Get();
+
+  const std::string cleaned_text =
+      embedding_proc_pipeline->CleanText(text, true);
+  if (cleaned_text.length() == 0) {
+    BLOG(1, "No text available for embedding");
+    return;
+  }
+
+  ml::pipeline::TextEmbeddingData embedding_data =
+      embedding_proc_pipeline->EmbedText(cleaned_text);
+  if (embedding_data.embedding.GetNonZeroElementsCount() == 0) {
+    BLOG(1, "Text not embedded");
```

Comment:
maybe "Failed to generated/create text embedding"

Reply:
Changed to "Failed to create text embedding"